packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/computeAxisValue.ts

```diff
@@ -208,6 +210,12 @@ export function computeAxisValue<T extends ChartSeriesType>({
       scale: finalScale.domain(domain) as any,
       tickNumber,
       colorScale: axis.colorMap && getColorScale(axis.colorMap),
+      valueFormatter:
+        axis.valueFormatter ??
+        (createScalarFormatter(tickNumber) as <TScaleName extends ScaleName>(
+          value: any,
+          context: AxisValueFormatterContext<TScaleName>,
+        ) => string),
```

Comment:
Need this cast because `createScalarFormatter` expects the scale to be `ContinuousScaleName`, but the type expects it to extend `ScaleName`. 

At this point, we know it can only be a continuous scale, but TypeScript doesn't know that.